File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    
    hasDuplicate(nums) {
        const hashMap = new Map()
        for (let i = 0; i < nums.length; i++) {
            if (hashMap.get(nums[i]) !== undefined) return true; 
            hashMap.set(nums[i], true);
        }
        return false
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> track = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            var diff = target - nums[i];
            if (track.get(diff) != null) {
                return new int[]{track.get(diff), i};
            } else {
                track.put(nums[i], i);
            }
            
        }
        return new int[]{0, 1};
    }
}
